### ISSUE TITLE
Make more files abort pre-commit if unstaged

### ISSUE THREAD

**Issue by ymyke** [pre-commit/pre-commit#1649]
`pre-commit` aborts if the config file (`.pre-commit-config.yaml`) is unstaged, and for good reason.

Is there a way to add more files to that same mechanic? 

E.g., it would be useful for `.pylintrc` for the same reason. Or also for some of the scripts I use in `local` hooks.

**Comment by asottile** [comment]
yes, the pre-commit configuration is special because _pre-commit_ cannot function without it

**Comment by ymyke** [comment]
I don't understand your answer (should the "yes" be a "no" in your answer?). 

So there is no way to add more files to the same mechanic (i.e., aborting pre-commit if they have unstaged changes)?

Any idea for a workaround?

**Comment by asottile** [comment]
correct, the answer is no

the pre-commit configuration is special because the framework cannot work without it, the rest of it is just passing filenames to tools

from a correctness standpoint, unstaging the pylintrc will still give you the correct results from a commit (it checks the contents as if they were trying to be committed) -- in your case correctly rejecting the commit because the configuration was not staged.  so I don't really see a reason to add complexity for something that's going to fail anyway.  additionally, aborting for arbitrary files forces a workflow which may be undesirable for some of your contributors (partial commits are after all an intentional feature of `git`)

you could maybe implement something using [`fail_fast`](https://pre-commit.com/#top_level-fail_fast) but it's probably not worth the trouble?

**Comment by ymyke** [comment]
Yep, most likely not worth the trouble.

Is there a way to find out if there are unstaged modifications in the repo while pre-commit is running? (It's not obvious to me since the tool stashes the modifications away before running.)

**Comment by asottile** [comment]
oh right you won't have access to them since they're stashed before things run 🤔 

I don't think there's any way to detect that currently
